ios/MaasCore/Sources/MaasComponents/Components/ButtonVariants.swift

```diff
@@ -0,0 +1,208 @@
+struct ColorfulButton: View {
+    
+    let colors: ButtonColors
+    let input: Button.InputType
+    
+    @Environment(\.isEnabled) var isEnabled
+    
+    var body: some View {
+        Button(
+            input.text,
+            icon: input.icon,
+            foregroundColor: colors.foreground.stateful(isActive: input.isLoading, isEnabled: isEnabled),
+            backgroundColor: colors.background.stateful(isActive: input.isLoading, isEnabled: isEnabled),
+            isSmall: input.isSmall,
+            isHugging: input.isHugging,
+            isLoading: input.isLoading,
+            action: input.action
+        )
+    }
+}
+
+struct ButtonColors {
+    let foreground: StatefulColor
+    let background: StatefulColor
+}
+
+struct StatefulColor {
+    let color: Color
+    let activeColor: Color
+    let disabledColor: Color
+    
+    func stateful(isActive: Bool, isEnabled: Bool) -> Color {
+        if !isEnabled { return disabledColor }
+        if isActive { return activeColor }
+        return color
+    }
+    
+    static func just(_ color: Color) -> StatefulColor {
+        StatefulColor(color: color, activeColor: color, disabledColor: color)
+    }
+}
+
+public struct PrimaryButton: View {
```

Comment:
Can the standard `Button` be "primary", i.e. have the primary button defaults?

Reply:
Standard button only has content and background colors, PrimaryButton has colors for disabled / active states as well

Reply:
I see 👍